lib/spack/spack/build_environment.py

```diff
@@ -302,6 +302,19 @@ def set_build_environment_variables(pkg, env, dirty=False):
             if '/macports/' in p:
                 env.remove_path('PATH', p)
 
+    # Set environment variables if specified for
+    # the given compiler
+    compiler = pkg.compiler
+    environment = compiler.environment
+    if 'set' in environment:
+        env_to_set = environment['set']
+        for key, value in env_to_set.iteritems():
+            env.set('SPACK_%s' % key, value)
```

Comment:
I think this (and the other comments) can be handled by modifying the environment variables in build_environment vs. passing things through to the compiler wrapper. It turns out this was originally done because the wrappers unset these variables.